### PR TITLE
#264: backend tests createWorkPackage Endpoint

### PR DESCRIPTION
**Pull request by anurag-arasan**
## Changes

_Explanation of changes goes here_

## Notes

_Any other notes go here_

## Test Cases

- Case A
- Edge case
- ...

## Screenshots

_If you made UI changes you must post a screenshot of the whole page for each change in 1) a normal sized window and 2) the smallest possible window_

_If you did any manual testing (e.g., with Postman), put screenshots of the http request and before and after of the db_

_If none of this applies, you can delete this section_

## To Do

_Any remaining things that need to get done_

- [ ] item 1
- [ ] ...

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #)

### CODE REVIEW

File: src/backend/tests/work-packages.test.ts [hunk 1]
```diff
@@ -91,4 +112,26 @@ describe('Work Packages', () => {
...
+
+  test('getChangeRequestReviewState returns false when changeRequest has not been reviewed', async () => {
+    jest.spyOn(prisma.user, 'findUnique').mockResolvedValue(batman);
+    jest.spyOn(prisma.change_Request, 'findUnique').mockResolvedValue(unreviewedCr);
+    const spy = jest.spyOn(Utils, 'getChangeRequestReviewState');
+
+    const res = await request(app).post('/create').send(createWorkPackagePayload);
+
+    expect(spy).toHaveBeenCalledWith(createWorkPackagePayload.crId);
+    expect(spy).toHaveReturnedWith(false);
```
Comment (anurag-arasan) [inline src/backend/tests/work-packages.test.ts:135]: idk why this isn't working, the actual values return as expected (from console.log testing), but jest is saying an empty object was returned.

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
you definitely have the right idea, just slightly off on how you're declaring/expecting stuff. Check out this pr https://github.com/Northeastern-Electric-Racing/FinishLine/pull/275

also pls:
* commit message should have the issue number
* branch name has the # sign (same with commit messages)
* fill out the pr form or delete sections that don't apply
* do the things in the PR checklist (like assigning yourself and linking the ticket to the pr)
* make sure there are no linting errors

File: src/backend/tests/work-packages.test.ts [hunk 2]
```diff
@@ -91,4 +112,26 @@ describe('Work Packages', () => {
     expect(res.statusCode).toBe(400);
     expect(res.body.message).toBe('One of the dependencies was not found.');
   });
+
+  test('getChangeRequestReviewState returns null when changeRequest is not found', async () => {
+    jest.spyOn(prisma.user, 'findUnique').mockResolvedValue(batman);
+    jest.spyOn(prisma.change_Request, 'findUnique').mockResolvedValue(null);
+    const spy = jest.spyOn(Utils, 'getChangeRequestReviewState');
+
+    const res = await request(app).post('/create').send(createWorkPackagePayload);
```
Comment (Ari-S-123) [inline src/backend/tests/work-packages.test.ts:121]: here

File: src/backend/tests/work-packages.test.ts [hunk 3]
```diff
@@ -91,4 +112,26 @@ describe('Work Packages', () => {
...
+    expect(spy).toHaveBeenCalledWith(createWorkPackagePayload.crId);
+    expect(spy).toHaveReturnedWith(null);
+  });
+
+  test('getChangeRequestReviewState returns false when changeRequest has not been reviewed', async () => {
+    jest.spyOn(prisma.user, 'findUnique').mockResolvedValue(batman);
+    jest.spyOn(prisma.change_Request, 'findUnique').mockResolvedValue(unreviewedCr);
+    const spy = jest.spyOn(Utils, 'getChangeRequestReviewState');
+
+    const res = await request(app).post('/create').send(createWorkPackagePayload);
```
Comment (Ari-S-123) [inline src/backend/tests/work-packages.test.ts:132]: and here

**Review by Ari-S-123 — CHANGES_REQUESTED** [review]
You have linting warnings.

**Pull request closed without merging**